### PR TITLE
Black Goo Patch

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
Patches Black Goo so there's no more random/accidental infections.

# Explain why it's good for the game
APPARENTLY we're randomly having zombies pop up
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
admin: Added a flag that determines if zombies can spread or not. Unauthorised spread will create a big admin message.
/:cl:

### CODE REVIEW

File: code/datums/diseases/mob_procs.dm [hunk 1]
```diff
@@ -7,6 +7,10 @@
 
 // This proc has some procs that should be extracted from it. I believe we can develop some helper procs from it - Rockdtben
 /mob/proc/contract_disease(datum/disease/virus, skip_this = 0, force_species_check=1, spread_type = -5)
+	if(istype(virus, /datum/disease/black_goo) && (SSticker.mode && !MODE_HAS_TOGGLEABLE_FLAG(MODE_ALLOW_ZOMBIES)))
```
Comment (fira) [inline code/datums/diseases/mob_procs.dm:10]: Forest seriously I swear.... OOP exists for a reason. At least do it 2018 style and define a handler proc on the disease, don't strap istypes based chunks in the middle of stuff like it's 2012

Comment (fira) [reply code/datums/diseases/mob_procs.dm:10]:
```
if(virus.can_infect(src))
```
or something i don't know

**Review by fira — CHANGES_REQUESTED** [review]

File: code/datums/diseases/mob_procs.dm (hunk 1 above)
Comment (realforest2001) [reply code/datums/diseases/mob_procs.dm:10]: But where's the fun in that. If I did everything perfectly first time around you wouldn't have so much fun reviewing it! :HAH:

Comment (fira) [reply code/datums/diseases/mob_procs.dm:10]: Mean :sob:

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/datums/diseases/mob_procs.dm (hunk 1 above)
Comment (realforest2001) [reply code/datums/diseases/mob_procs.dm:10]: Will do this when/if I have time tomorrow

**Pull request closed without merging**